streams/shared/src/main/scala/zio/stream/ZPipeline.scala

```diff
@@ -1126,7 +1126,7 @@ object ZPipeline extends ZPipelinePlatformSpecificConstructors {
                 acc ++ Chunk.fromArray(bytes)
               }
 
-              ZChannel.write(bytes)
+              ZChannel.write(bytes).zipRight[Any, Nothing, Chunk[String], Any, Nothing, Chunk[Byte], Any](transform)
```

Comment:
Is ` *> transform` not typechecking? If it is, that would look cleaner

Reply:
```scala
new ZPipeline(ZChannel.write(bom).zipRight(transform))
```
[error] /zio/streams/shared/src/main/scala/zio/stream/ZPipeline.scala:1129:46: type mismatch;
found   : zio.stream.ZChannel[Any,Nothing,zio.Chunk[String],Any,Nothing,zio.Chunk[Byte],Any]
[error]  required: zio.stream.ZChannel[Any,InErr1,zio.Chunk[String],Any,Nothing,zio.Chunk[Byte],Any]
[error]               ZChannel.write(bytes).zipRight(transform)


I followed your instruction, but it occurred error

Reply:
```scala
new ZPipeline(ZChannel.write(bom).zipRight(transform))
```
[error] /zio/streams/shared/src/main/scala/zio/stream/ZPipeline.scala:1129:46: type mismatch;
found   : zio.stream.ZChannel[Any,Nothing,zio.Chunk[String],Any,Nothing,zio.Chunk[Byte],Any]
[error]  required: zio.stream.ZChannel[Any,InErr1,zio.Chunk[String],Any,Nothing,zio.Chunk[Byte],Any]
[error]               ZChannel.write(bytes).zipRight(transform)


I followed your instruction, but it occurred error